### ISSUE TITLE
Indent should ignore indentation of variable assigment in multiline variable declaration

### ISSUE THREAD

**Issue by soda0289** [eslint/eslint#8546]
**Tell us about your environment**
* **ESLint Version:**
master
* **Node Version:**
6.10
* **npm Version:**
3.10

**What parser (default, Babel-ESLint, etc.) are you using?**
typescript
**Please show your full configuration:**

```js
module.exports = {
//    "extends": ["eslint:all"],
    "parser": "typescript-eslint-parser",
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "rules": {
        "indent": "error"
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

Invalid
```js
const fooLongName
        = 1,
    barLongName
        = 2,
    bazLongName
        = 3;
```

**What did you expect to happen?**

I expected this to be a valid indentation style. The indent-legacy rule allows for this style. Also if equal operators are on the previous line it is considered a valid indentation.

Valid
```js
const fooLongName =
        1,
    barLongName =
        2,
    bazLongName =
        3;
```

**What actually happened? Please include the actual, raw output from ESLint.**
It is expected to be indented like so:

```js
const fooLongName
    = 1,
    barLongName
    = 2,
    bazLongName
    = 3;
```

```
/mnt/c/Users/reyad.attiyat/Workspace/eslint-test/test-var-indent.ts
  2:1  error  Expected indentation of 4 spaces but found 8  indent
  4:1  error  Expected indentation of 4 spaces but found 8  indent
  6:1  error  Expected indentation of 4 spaces but found 8  indent
```

I can a create a PR to fix this. Just want to confirm this is not intended behavior. When the equal operator is on the line with the variable identifier the assignment indentation is ignored completely. We should probably do the same when the equal operator is on the same line as the assignment.

I have tested the changes made in https://github.com/eslint/eslint/pull/8498 and it does not fix this case.

**Comment by not-an-aardvark** [comment]
This seems like a bug to me, I suppose. The rule ignores the indentation of the equals signs in assignments, so it makes sense to be consistent with variable declarations.

**Comment by soda0289** [comment]
I will work on a fix.
This is the last issue I have with the new rule. Everything else works much better than the old indent rule!
